src/arch/loong64/Makefile

```diff
@@ -8,11 +8,20 @@ ASM_TCHAR_OPS	:= @
 CFLAGS		+= -fstrength-reduce -fomit-frame-pointer
 CFLAGS		+= -falign-jumps=1 -falign-loops=1 -falign-functions=1
 
+# Prevent use of lsx and lasx registers
+#
+CFLAGS		+= -mno-lsx -mno-lasx
+
 # Check if -mno-explicit-relocs is valid
 ifeq ($(CCTYPE),gcc)
 MNER_TEST = $(CC) -mno-explicit-relocs -x c -c /dev/null -o /dev/null >/dev/null 2>&1
 MNER_FLAGS := $(shell $(MNER_TEST) && $(ECHO) '-mno-explicit-relocs')
 WORKAROUND_CFLAGS += $(MNER_FLAGS)
+
+# Check if -mno-relax is valid
+MNRX_TEST = $(CC) -mno-relax -x c -c /dev/null -o /dev/null >/dev/null 2>&1
+MNRX_FLAGS := $(shell $(MNRX_TEST) && $(ECHO) '-mno-relax' || $(ECHO) '-Wa,-mno-relax' )
```

Comment:
This looks wrong: we check for support of `-mno-relax`, but then apply `-Wa,-mno-relax`.  Why is this different?